### ISSUE TITLE
[Test failure] TC_AUTHOR_REDWOOD_42: Deleting third level child tags shouldn't remove second level child tags

### ISSUE THREAD

**Issue by cassiezamparini** [openedx/wg-build-test-release#378]
### Release

Redwood

### Expected behavior

When I delete a third level child tag, the second level child tag shouldn't delete as well.

### Actual behavior

See this video for the actual behaviour: https://www.loom.com/share/aea0455461b34bc98732c77805f46aa3?sid=1cafa5ad-7be0-4c6c-9658-7a43a71783c5

### Steps to reproduce

1. Add a parent tag 
2. Add a second level and third level child tag
3. Click save
4. Reopen the tag drawer and delete the third level tag. The second level tag will disappear as well. This should not happen.

**Comment by cassiezamparini** [comment]
cc. @jmakowski1123 FYI

**Comment by cassiezamparini** [comment]
@jmakowski1123 @ali-hugo and I were chatting about this, so just to confirm...

1. The Level 2 tag shouldn't delete if I've explicitly selected it (if I delete a level 3 tag) as mentioned above.
2. BUT if I **_don't_** explicitly select a Level 2 tag, but explicitly select a Level 3 tag and then delete it, the Level 2 tag should automatically delete. It works in this way already.

**Comment by ali-hugo** [comment]
@cassiezamparini Although I agree that the fact that "chordophone" disappears feels strange, I don't think it's actually a bug. I found [this ticket](https://github.com/openedx/modular-learning/issues/122) describing the functionality. See the following points:

- 2. Display an "x" for explicit tags in the tag drawer. Clicking the "x" will remove the tag from the content item.
- 5.5. The most granular tag in the hierarchy becomes the explicit tag with all its parent tags becoming implicit
- 3. Removing an explicit child that is the only child of its implicit parent will remove the implicit parent too (and grandparent etc.)

Let me know if you agree that the actual behaviour is in line with what is described in that ticket.

**Comment by cassiezamparini** [comment]
@ali-hugo Yes, it seems to be inline with these two points especially: _The most granular tag in the hierarchy becomes the explicit tag with all its parent tags becoming implicit; Removing an explicit child that is the only child of its implicit parent will remove the implicit parent too (and grandparent etc.)_ Thanks Ali.

Confusing though. I agree we need to do some work on the overall UX here. There were one or two things I was confused by during testing again.

@jmakowski1123 Should I archive this issue? Anything you'd like to add?

**Comment by jmakowski1123** [comment]
> @cassiezamparini Although I agree that "chordophone" disappearing feels strange, I don't think it's actually a bug. I found [this ticket](https://github.com/openedx/modular-learning/issues/122) describing the intended functionality. See the following points:
> 
> 2. Display an "x" for explicit tags in the tag drawer. Clicking the "x" will remove the tag from the content item.
> 
> 5.5. The most granular tag in the hierarchy becomes the explicit tag with all its parent tags becoming implicit
> 
> 3. Removing an explicit child that is the only child of its implicit parent will remove the implicit parent too (and grandparent etc.)
> 
> Let me know if you agree that the actual behaviour is in line with what is described in that ticket.
> 
> Hopefully we'll get the opportunity to improve the tagging user experience in the future.

@cassiezamparini @ali-hugo I think the bug is of a different nature then. This is the intended behavior:

_Removing an explicit child that is the only child of its implicit parent will remove the implicit parent too (and grandparent etc.)_

But in the example video Cassie posted, when the grandchild tag (piano) was deleted, chordophone was also deleted, but instrument was not. In the story, I would assume that chordophone and instrument would both become implicit tags. Do you interpret it the same way?

**Comment by ali-hugo** [comment]
@jmakowski1123 

> In the story, I would assume that chordophone and instrument would both become implicit tags. Do you interpret it the same way?

I do. 

It's strange though, because it looks like "percussion instruments" _does_ become implicit after @cassiezamparini selects the "add" button (i.e. there is no longer a grey "x" next to it), but it is not treated as an implicit tag when its children tags are deleted (i.e. it does not get deleted).

**Comment by mariajgrimaldi** [comment]
@jmakowski1123 @cassiezamparini @ali-hugo: could you confirm whether this is the intended behavior? Thanks!

**Comment by cassiezamparini** [comment]
@jmakowski1123 So do you agree with the updated description here: https://github.com/openedx/wg-build-test-release/issues/378#issue-2334850863
